### PR TITLE
Print all CTS outputs (from gtest) in CI.

### PR DESCRIPTION
**Pull request by szadam**
An example log:
https://github.com/szadam/unified-runtime/actions/runs/6629114965/job/18007695019?pr=8

### CODE REVIEW

**Review by lukaszstolarczuk — APPROVED** [review]
Can you please paste here some example log?

**Review by pbalcer — COMMENTED** [review]
This seems fairly elaborate. Would it be possible to allow `stdout` to pass-through and print just the filtered lines to `stderr`? Then we could change the match function to use `stderr` instead (https://github.com/oneapi-src/unified-runtime/blob/adapters/test/conformance/CMakeLists.txt#L16).

Should be simpler, and the `cts_exe.py` script would become transparent with regards to output.

File: test/conformance/cts_exe.py [hunk 1]
```diff
@@ -20,16 +22,21 @@
     parser.add_argument("--test_command", help="Ctest test case")
 
     args = parser.parse_args()
-    result = subprocess.Popen([args.test_command, '--gtest_brief=1'], stdout = subprocess.PIPE, text = True)  # nosec B603
+    output_file = open("output.txt", "w")
```
Comment (PatKamin) [inline test/conformance/cts_exe.py:25]:
```
with open() as output_file:
```
Then you can remove the closing file command as the file will be closed at the end of the `with` clause scope.

File: scripts/read_cts_output.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ See LICENSE.TXT
+ SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""
+# Print conformance test output from gtest, which is captured by the cts_exe.py script.
+
+import os
+import sys
+
+def print_file_content(folder_path, conformance_folder):
```
Comment (PatKamin) [inline scripts/read_cts_output.py:14]: `conformance_folder` is the last part of the `folder_path`. You can get it like this I suppose:
```
os.path.basename(folder_path)
```
Then there would be no need to pass it as the argument to this function.

**Comment by pbalcer** [comment]
The changes in #832 seem to be superseding this PR. Can we close it?

File: test/conformance/README.md [hunk 3]
```diff
@@ -8,4 +8,4 @@ In the future, when all bugs are fixed, and the tests pass,
 this solution will no longer be necessary.
 When you fix any test, the match file must be updated
 Empty match files indicate that there are no failing tests
-in a particular group for the corresponding adapter.
+in a particular group for the corresponding adapter.
```
Comment (lukaszstolarczuk) [inline test/conformance/README.md:11]: is there any relevant change here? (perhaps GH is not showing me something)

File: test/conformance/cts_exe.py [hunk 4]
```diff
@@ -18,18 +20,24 @@
...
             print(test_case)
 
-    result.communicate()
-    rc = result.returncode
+    rc = result.wait()
     if rc < 0:
-        print(signal.strsignal(abs(result.returncode)))
+        print(signal.strsignal(abs(rc)))
+
+    print("#### OUTPUT ####", file=sys.stderr)
```
Comment (lukaszstolarczuk) [inline test/conformance/cts_exe.py:40]: pls mark here, this output is from gtest

**Review by kbenzie — APPROVED** [review]
I always appriciate more context in CI failures.

File: test/conformance/README.md (hunk 3 above)
Comment (szadam) [reply test/conformance/README.md:11]: space at the end of sentence.

File: test/conformance/cts_exe.py (hunk 4 above)
Comment (szadam) [reply test/conformance/cts_exe.py]: Done

**Review by lukaszstolarczuk — APPROVED** [review]

File: scripts/read_cts_output.py (hunk 2 above)
Comment (szadam) [reply scripts/read_cts_output.py]: outdated

File: test/conformance/cts_exe.py (hunk 1 above)
Comment (szadam) [reply test/conformance/cts_exe.py]: outdated

**Review by PatKamin — APPROVED** [review]

**Review by pbalcer — APPROVED** [review]

**Pull request merged**